src/plugins/console/public/application/hooks/use_send_current_request_to_es/use_send_current_request_to_es.ts

```diff
@@ -46,29 +48,37 @@ export const useSendCurrentRequestToES = () => {
       const results = await sendRequestToES({ requests });
 
       let saveToHistoryError: undefined | Error;
+      const { historyDisabled } = settings.toJSON();
 
-      results.forEach(({ request: { path, method, data } }) => {
-        try {
-          history.addToHistory(path, method, data);
-        } catch (e) {
-          // Grab only the first error
-          if (!saveToHistoryError) {
-            saveToHistoryError = e;
+      if (!historyDisabled) {
+        results.forEach(({ request: { path, method, data } }) => {
+          try {
+            history.addToHistory(path, method, data);
+          } catch (e) {
+            // Grab only the first error
+            if (!saveToHistoryError) {
+              saveToHistoryError = e;
+            }
           }
-        }
-      });
+        });
+      }
 
       if (saveToHistoryError) {
         const errorTitle = i18n.translate('console.notification.error.couldNotSaveRequestTitle', {
           defaultMessage: 'Could not save request to Console history.',
         });
         if (isQuotaExceededError(saveToHistoryError)) {
-          notifications.toasts.addError(saveToHistoryError, {
-            title: errorTitle,
-            toastMessage: i18n.translate('console.notification.error.historyQuotaReachedMessage', {
+          notifications.toasts.addWarning({
+            title: i18n.translate('console.notification.error.historyQuotaReachedMessage', {
               defaultMessage:
-                'Request history is full. Clear the Console history to save new requests.',
+                'Request history is full. Clear the console history or disable saving new requests.',
             }),
+            text: toMountPoint(
+              StorageQuotaError({
+                onClearHistory: () => history.clearHistory(),
+                onDisableSavingToHistory: () => settings.setHistoryDisabled(true),
```

Comment:
Once the history has been disabled, there's no where in the UI for the user to re-enable it again. What do you think about adding a toggle in the console settings modal to allow the users to change this setting?

Reply:
Hi @sabarasaba , Thanks for taking the time to review this PR, I made the suggested changes